Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -0,0 +1,70 @@
+namespace gdjs {
+  const logger = new gdjs.Logger('Leaderboards');
+  export namespace evtTools {
+    export namespace leaderboards {
+      let _scoreLastSentAt: number | null = null;
+      let _lastScore: number;
+      let _lastPlayerName: string;
+      let _lastErrorCode: number;
+
+      export const setPlayerScore = function (
+        leaderboardId: string,
+        score: float,
+        playerName: string,
+        responseVar: gdjs.Variable,
+        errorVar: gdjs.Variable
+      ) {
+        errorVar.setString('');
+        responseVar.setString('');
+
+        if (
+          (_lastPlayerName === playerName && _lastScore === score) ||
+          (!!_scoreLastSentAt && Date.now() - _scoreLastSentAt < 500)
+        ) {
+          errorVar.setString('Wait before sending a new score.');
+        } else {
+          const baseUrl = 'https://api.gdevelop-app.com/play';
+          fetch(
+            `${baseUrl}/game/${gdjs.projectData.properties.projectUuid}/leaderboard/${leaderboardId}/entry`,
+            {
+              body: JSON.stringify({
+                playerName: playerName,
```

Comment:
I think we could send playerUuid by exposing `_makePlayerUuid` from runtimeGame (that you can get from the runtime scene)